cpp/src/unary/math_ops.cu

```diff
@@ -264,6 +282,7 @@ std::unique_ptr<column> unary_op_with(column_view const& input,
   using Type                     = device_storage_type_t<T>;
   using FixedPointUnaryOpFunctor = FixedPointFunctor<Type>;
 
+  // When scale is >= 0 and unary_op is CEIL or FLOOR, the unary_operation for is a no-op
   if (input.type().scale() >= 0 &&
       not std::is_same<FixedPointUnaryOpFunctor, fixed_point_abs<Type>>::value)
```

Comment:
It would be safer to actually have ceil and floor in the std::is_same, right? When someone comes along and adds `fixed_point_negate`, for example, they would need to recognize this doesn't work for them and add some code here, but if it were to call out the ceil and floor ops now it would work no matter what was added and the optimization would be opt-in.

Reply:
100% agree. Had the same though myself but i figured that I would be the one to modify this code next so it mattered less - but that is not a good reason. Fixed.